### PR TITLE
hf compatibility test returns value + cleanup

### PR DESCRIPTION
**Pull request by KUNAL1612**
**Patch Description**
Owing to the changes in #264, there is now no longer a need to save the trimmed logits to an external file instead we could just return it directly. Also cleaned up a few unnecessary lines of code.

### CODE REVIEW

**Review by punitkoura — APPROVED** [review]
Thanks for cleaning this up!
